Refactor Wordle Input Handling and Add Dynamic Game Feedback Features

andreuu-tsai opened the pull request:
### Summary
This pull request introduces the following updates to the Wordle game component and its related modules:

---

Feature: 
#3 

### **Key Changes**

#### 1. **Refactor Input Handling**
- **Removed `useWordleInput` hook**:
  - Integrated its logic directly into the Wordle component to simplify state management.
  - Eliminated unnecessary abstraction and made the input handling logic clearer and easier to maintain.
  
- **Enhanced `handleInput` Logic**:
  - Moved the `ENTER` key submission logic outside of `setInputValue` to prevent duplicate submissions caused by state changes.

- **Cleaned Up Redundant Code**:
  - Removed the `resetInput` function and replaced it with a direct state reset (`setInputValue("")`).
  - Removed unnecessary `console.log` statements.

---

#### 2. **Dynamic Feedback on Game Results**
- **Added New `generateEncouragementOrTaunt` Function**:
  - Dynamically generates encouraging or humorous messages based on the user's guesses and whether they lost the game.
  - Uses GPT-based text generation (`ChatOpenAI`) to provide interactive and entertaining messages.

- **Integrated Feedback into `submitWord`**:
  - Triggers `generateEncouragementOrTaunt` when the user loses, providing a personalized message about their performance.

andreuu-tsai committed:
feat(chat): add encouragement or taunt generation for Wordle game

- Implemented `generateEncouragementOrTaunt` function in `chat.ts` to generate dynamic messages based on game outcomes.
- Integrated the new function into `submitWord` in `wordle-actions.ts` to provide user feedback when the game is lost.

andreuu-tsai committed:
Refactor Wordle component: Simplify input handling and remove useWordleInput hook

- Removed `useWordleInput` custom hook and integrated its logic directly into the Wordle component.
- Refactored `handleInput` to manage key input handling, submission, and input resetting within the component.
- Moved the `ENTER` key submission logic outside of `setInputValue` to prevent duplicate submissions caused by state changes.
- Introduced `useState` to handle `inputValue` directly in the Wordle component.
- Removed unnecessary `resetInput` logic in favor of clearing `inputValue` state directly.
- Cleaned up `wordle-actions.ts` by removing a redundant `console.log` statement.

EricTsai83 commented on src/lib/chat.ts:
```diff
@@ -25,6 +25,36 @@ export default async function generateCongrats(
+  addMessage(userId, "mii", content);
+}
+
 export async function addMessage(
```
Authenticate first before connecting to the database.

EricTsai83 reviewed (changes requested):
Make modifications based on the comment.

andreuu-tsai committed:
feat: add authentication checks to server actions

- Introduced authenticate function in utils.ts to verify user sessions.
- Added authenticate calls to the following functions for user authorization:
  - generateCongrats
  - generateEncouragementOrTaunt
  - addMessage
  - getMessagesByUserId
  - getGameByUserId
  - submitWord
  - newGame
  - resetGame
- Ensured unauthorized access throws an error when no session or mismatched user ID is detected.

andreuu-tsai commented:
Done adding the authentication at server actions. Please review again.

EricTsai83 reviewed (approved):
LGTM
